### ISSUE TITLE
dbt-0.21.0 test issue: Alias getting added in test column which is causing test failure

### ISSUE THREAD

**Issue by Ruchika2507** [dbt-labs/dbt-bigquery#52]
### Describe the bug
Command executed: ` dbt test --model <model name>`
Error Message: 
```
   SELECT list expression references column <column_name> which is neither grouped nor aggregated at [12:5]
      compiled SQL <location>\unique_test_model_ID_Name.sql
```

### Steps To Reproduce
To reproduce execute following commands:

1. Run model:  `dbt run --model test_model`
2. Test model: `dbt test --model test_model`

**test_model.sql**
```
  Select 
    '1' as ID
    , 'abc' as Name
```

**mart.yml**
```
version: 2

models:
  - name: test_model
    columns:
      - name: ID || Name
        tests:
          - unique
```

### Expected behavior
Test case should have executed successfully.

### System information
**The output of `dbt --version`:**
```
0.21.0
```

### Additional context
The test case was running fine with dbt version: 0.19.0

### Compiled SQL with dbt 0.21.0

```
select
      count(*) as failures,
      count(*) != 0 as should_warn,
      count(*) != 0 as should_error
    from (
            select
                ID || Name as unique_field,
                count(*) as n_records
            from `<project_id>`.`<schema>`.`test_model`
            where ID || Name is not null
            group by ID || Name
            having count(*) > 1
    ) dbt_internal_test
```

### Compiled SQL with dbt 0.19.0
```
select count(*) as validation_errors
from (

    select
        ID || Name

    from `<project_id>`.`<schema>`.`test_model`
    where ID || Name is not null
    group by ID || Name
    having count(*) > 1

) validation_errors

```

### ISSUE:
Seems the aliasing of the OR field creates an issue for BigQuery.

**Comment by jtcohen6** [comment]
@Ruchika2507 Thanks for the detailed bug report. I believe this is a duplicate of https://github.com/dbt-labs/dbt-bigquery/issues/35, which we fixed in #10. That fix is going into v1, though it's not currently planned to go into v0.21.1.

The issue is, unlike the majority of other databases, BigQuery does not support grouping by a column expression if that column expression is also aliased.

In the meantime, you can confirm that's the issue, and access the fix, by copying the contents of the new, BigQuery-specific `unique` test SQL into your own project:
https://github.com/dbt-labs/dbt-bigquery/blob/bb9caab8fd771d72bafdcfffb1ae72459ad704a2/dbt/include/bigquery/macros/adapters.sql#L198-L216

I'm going to close this for now, but I can reopen if we determine there's a different underlying issue.

**Comment by Ruchika2507** [comment]
Thanks @jtcohen6.
